roles/network_plugin/multus/templates/multus-daemonset.yml.j2

```diff
@@ -1,8 +1,9 @@
+{% for container_manager_type in container_manager_types %}
 ---
 kind: DaemonSet
 apiVersion: apps/v1
 metadata:
-  name: kube-multus-ds-amd64
+  name: kube-multus-{{ container_manager_type }}-amd64
```

Comment:
You can walk `hostvars` ansible variable instead and detect all possible container_manager versions configured and render a file for each one instead of doing this.

Reply:
Indeed, I’ll try to check this way. Thanks.

Reply:
I was added exactly a range in template to get one manifest file but for different container engine types. So, this multi daemonsets manifests will be applied as separated one for each group of worker nodes with differ container engine type.
Also do  you mean to get all types inside templete level without preparing a variable before on task level ?

Reply:
It is a good practice to have one yaml manifest per file, this translates into one manifest file for each daemonset.

Reply:
I can rework it to use different files and will try to play with hosts vars, ok. I’ll do that by tomorrow.